ipatests/test_integration/test_ntp_options.py

```diff
@@ -0,0 +1,196 @@
+        assert server_install.returncode == 2
+        assert exp_str in server_install.stderr_text
+        args2 = ['ipa-client-install', '-N', '--ntp-server=%s' % ntp_server]
+        client_install = self.client.run_command(args2, raiseonerr=False)
+        assert client_install.returncode == 2
+        assert exp_str in client_install.stderr_text
+        args3 = ['ipa-client-install', '-N',
+                 '--ntp-pool=%s' % ntp_server.lstrip('1.')]
+        client_install = self.client.run_command(args3, raiseonerr=False)
+        assert client_install.returncode == 2
+        assert exp_pool_str in client_install.stderr_text
+
+    def test_two_step_replica_install_using_ntp_options(self):
+        """
+        test to verify 2-step ipa-replica install with --ntp-server, --ntp-pool
+        and -N options
+        """
+        ntp_server = "1.pool.ntp.org"
+        ntp_pool = "pool.ntp.org"
+        exp_str = "NTP configuration cannot be updated during promotion"
+        tasks.install_master(self.master, setup_dns=False)
+        tasks.install_client(self.master, self.replica)
+        replica_install = self.replica.run_command(
+            ['ipa-replica-install', '-N'], raiseonerr=False)
+        assert replica_install.returncode == 1
+        assert exp_str in replica_install.stderr_text
+        replica_install = self.replica.run_command(
+            ['ipa-replica-install', '--ntp-server=%s' % ntp_server],
+            raiseonerr=False)
+        assert replica_install.returncode == 1
+        assert exp_str in replica_install.stderr_text
+        replica_install = self.replica.run_command(
+            ['ipa-replica-install', '--ntp-pool=%s' % ntp_pool],
+            raiseonerr=False)
+        assert replica_install.returncode == 1
+        assert exp_str in replica_install.stderr_text
+        tasks.uninstall_master(self.replica)
+        self.cleanup()
+
+    def test_two_step_replica_install_without_ntp_options(self):
+        """
+        test to verify 2-step ipa-replica install
+         - ipa-client install with ntp option
+         - ipa-replica without ntp option
+        """
+        ntp_pool = "pool.ntp.org"
+        exp_str = "ipa-replica-install command was successful"
+        tasks.install_master(self.master, setup_dns=False)
+        tasks.install_client(self.master, self.replica,
+                             extra_args=['--ntp-pool=%s' % ntp_pool])
+        replica_install = self.replica.run_command(
+            ['ipa-replica-install'], raiseonerr=False)
+        assert exp_str in replica_install.stderr_text
+        cmd = self.replica.run_command(['cat', paths.CHRONY_CONF])
+        assert ntp_pool in cmd.stdout_text
+        tasks.uninstall_master(self.replica)
+
+    def cleanup(self):
+        """
+        Uninstall ipa-server and ipa-client
+        """
+        tasks.uninstall_client(self.client)
+        tasks.uninstall_master(self.master)
```

Comment:
@varunmylaraiah  what happens if we add replica uninstallation to cleanup as well?
```suggestion
        tasks.uninstall_replica(self.replica)
        tasks.uninstall_master(self.master)
```

Reply:
The question is meant mostly for tests that are not using replica.

Reply:
yes, all tests are not using replica install and replica uninstallation will take almost 45 seconds to 1 minute.
I think it's of no use for uninstalling replica every time.